crates/gen-host-js/src/lib.rs

```diff
@@ -1253,7 +1293,18 @@ impl<'a> JsInterface<'a> {
         self.src.ts("): ");
         match func.results.len() {
             0 => self.src.ts("void"),
-            1 => self.print_ty(func.results.iter_types().next().unwrap()),
+            1 => {
+                if func.results.throws(self.iface) {
```

Comment:
I'd recommend changing the signature of `throws` from returning a `bool` to returning `Option<(&Type, &Type)>` so you can avoid the `iter_types` and `if let` here to have it all bundled-in-one.